### ISSUE TITLE
Please provide the build script

### ISSUE THREAD

**Issue by stangri** [topjohnwu/MagiskManager#32]
Many thanks for the easy-peasy build script for magisk, could you please provide the same for Magisk Manager (at least for Linux/macOS)? There're still people with repo/ZIP installation issues, they would greatly benefit from the most recent Magisk Manager build.
Thanks!

**Comment by d8ahazard** [comment]
Lol wut?

https://lmgtfy.com/?q=Gradle

**Comment by stangri** [comment]
@d8ahazard if your teachers laughed out loud at your questions and that gave an impression that it's an acceptable social behavior -- I feel sorry for you.

PS. You may want to obfuscate lmgtfy links in the future if you want people to click on them. They even have the nice "shorten" feature just for that.

**Comment by topjohnwu** [comment]
@stangri  Build with Android Studio

**Comment by stangri** [comment]
@topjohnwu I've submitted a PR for modified build.gradle script which allows successful build from CLI.

**Comment by d8ahazard** [comment]
Dude.  Gradle is an automated tool.  You don't modify the build script, studio does.  You either compile the app with Android studio, or look up the command-line parameters to build it manually while ignoring lint.  

You can get all butthurt because I posted a reply suggesting you read up on how Gradle works before submitting issues and pull requests, but clearly, you didn't read the link.  

http://stackoverflow.com/questions/21143556/avoid-lint-when-gradle-execute-check

**Comment by stangri** [comment]
> You can get all butthurt
Classy.

> a reply suggesting you read up on how Gradle works before submitting issues and pull requests
We must have a very different understanding of the word "suggesting" and "read up". 

> clearly, you didn't read the link
**The link**? The google search on word gradle is **the link**? Was I supposed to go thru ~4m results? The build.gradle edit is what came out of searching for the lint error actually, I didn't come up with it on my own.

> http://stackoverflow.com/questions/21143556/avoid-lint-when-gradle-execute-check
Thanks, this link actually helped, PR is redundant of course. 

Since you seem to be expert in apk-related matters, any idea why with:
`java -jar signapk.jar MagiskManager/app/src/main/assets/public.certificate.x509.pem MagiskManager/app/src/main/assets/private.key.pk8  MagiskManager/app/build/outputs/apk/app-release-unsigned.apk ./MagiskManager-v2.6-snapshot.apk` I still get the signature mismatch over @topjohnwu's Magisk Manager 2.5?

PS. I would have gladly used Magisk v10 and MagiskManager v2.6 snapshots built and shared by someone else, with more knowledge, I'm only doing that because nobody else was doing that (at least the sharing part).

**Comment by d8ahazard** [comment]
I *am* classy.  I got me one of those tuxedo tee-shirts an everthing!

Those keys appear to be for signing the .zip files downloaded by MagiskManager from github.  While you can still use them to sign the APK, this explains the issue.  The keys used by John to sign and release the official versions of MagiskManager are his own private keys.  You can't get them, I can't get them, etc.  So, you can still compile, sign, and install your own 100% valid copy of the apk, but it's signature will ALWAYS mismatch the offical one.  Different signing keys, different signature.  This is nothing to be worried about, you will just have to uninstall and install the new version.

**Comment by stangri** [comment]
Thanks for your prompt reply!

> Those keys appear to be for signing the .zip files downloaded by MagiskManager from github. 

Ah, that reason escaped me!

**Comment by d8ahazard** [comment]
And now that I'm thinking of it, if the signature thing really really bugs you, you can do what everybody else does when modifying APK's and do the following:

1.  Open the original, proper apk of Magisk with winrar or 7zip.  Yeah, it's basically just a zip file.
2.  Open up the compiled version of the apk with the same program.  Extract everything to a folder.
3.  Delete AndroidManifest.xml.  This is the only file you cannot ever ever ever put into a stock signed apk.  Doing so breaks signatures.
4.  Drop the extracted files into the original opened zip, replace the ones currently there. 

As long as no new changes were added to Manifest from the last release, this will always work.  If the manifest was updated and new Activities, Listeners, whatever were added or removed from the program, then you can't use this method.  

Basically - don't change the manifest.  Don't try to create a new zip.  Use the existing apk and manifest, replace the .dex file, resources.arsc, and whatever's in the other folders.

Happy hacking...

**Comment by stangri** [comment]
Thanks man! 

I didn't want to reply until I've tried but as I started dragging my feet I realized it may look like I'm an unappreciative SOB, which (hopefully) I am not.

**Comment by stangri** [comment]
Didn't work, the following: `cd MagiskManager
./gradlew build -x lint
cp app/build/outputs/apk/app-release-unsigned.apk ../APKs/
cd ../APKs
unzip app-release-unsigned.apk -d new -x AndroidManifest.xml
cp MagiskManager-v2.5.apk MagiskManager-v2.6-snapshot.apk
cd new
zip -r ../MagiskManager-v2.6-snapshot.apk *`
results in a corrupted apk.
